name: 'Robot 3'
spin_rate: 1.0           # in Hertz
broadcast_rate: 0.7     # in Hertz
timeout_duration: 3.0    # in seconds
buffer_horizon: 8.0      # in seconds
sensors:
  size: 2
  sensor0:
    plugin_name: 'alliance_test/odometry'
    topic_name: 'odom'
    timeout_duration: 5.0
    buffer_horizon: 6.0
  sensor1:
    plugin_name: 'alliance_test/point_cloud'
    topic_name: 'sonar'
    timeout_duration: 6.0
    buffer_horizon: 5.0
behaviour_sets:
  size: 3
  behaviour_set0:
    task_id: 'wander'
    task_expected_duration: 225.0 # in seconds
    motivational_behaviour:
      threshold: 360.0
      acquiescence:
        yielding_delay: 150.0   # in seconds
        giving_up_delay: 250.0  # in seconds
      impatience:
        fast_rate: 5.0
      sensory_feedback:
        plugin_name: 'alliance_test/updated_sensory'
        sensors:
          size: 1
          sensor0:
            topic_name: 'sonar'
  behaviour_set1:
    task_id: 'border_protection'
    task_expected_duration: 325.0 # in seconds
    motivational_behaviour:
      threshold: 200.0
      acquiescence:
        yielding_delay: 100.0   # in seconds
        giving_up_delay: 300.0  # in seconds
      impatience:
        fast_rate: 10.0
      sensory_feedback:
        plugin_name: 'alliance_test/updated_sensory'
        sensors:
          size: 2
          sensor0:
            topic_name: 'odom'
          sensor1:
            topic_name: 'sonar'
  behaviour_set2:
    task_id: 'report'
    task_expected_duration: 60.0 # in seconds
    motivational_behaviour:
      threshold: 50.0
      acquiescence:
        yielding_delay: 60.0   # in seconds
        giving_up_delay: 50.0  # in seconds
      impatience:
        fast_rate: 7.5
      sensory_feedback:
        plugin_name: 'alliance_test/updated_sensory'
        sensors:
          size: 2
          sensor0:
            topic_name: 'odom'
          sensor1:
            topic_name: 'sonar'
